pkg/nfd-master/nfd-master.go

```diff
@@ -812,6 +817,9 @@ func filterExtendedResource(name, value string, features *nfdv1alpha1.Features)
 	// Check if given NS is allowed
 	ns, _ := splitNs(name)
 	if ns != nfdv1alpha1.ExtendedResourceNs && !strings.HasSuffix(ns, nfdv1alpha1.ExtendedResourceSubNsSuffix) {
+		if ns == "" {
```

Comment:
you can move this check just after `splitNs(name)`

Reply:
Maybe this is more readable, changed as suggested